### DISCUSSION TITLE
Vert.X clearing out Logging MDC

### DISCUSSION THREAD

**Discussion by bcluap** [quarkusio/quarkus#27254] [Q&A]
Since 2.10.X, it seems like slf4j MDC is being cleared behind the scenes by Vert.X see https://github.com/quarkusio/quarkus/pull/25311

I have a scenario where I add to MDC in a resteasy reactive request (Vertx context is there and on executor thread as its set as blocking) and then use the Vertx HTTP client which uses the eventloop and this eventloop thread has my MDC (great!). The calls to vertx HTTP client finish and my executor has been blocked waiting for the result.  But next thing the MDC disappears because the  eventloop thread used by the HTTP client has gone off and done something else and decided to clear my MDC. 

I think that integrating MDC with Vertx is an issue in this case. Perhaps the handoff to vertx HTTP client should create a new MDC and copy to it as opposed to referring to my MDC? Just wondering if there are thoughts around solving this.

**Comment by tugcesahin** [comment]
I have same problem since I upgraded Quarkus version from `2.10.4.Final` to `2.11.1.Final`. When I debug on `2.10.4.Final` version, `org.jboss.logmanager.ThreadLocalMDC` is used as `MDCProvider` but with `2.11.1.Final`, `io.quarkus.vertx.core.runtime.VertxMDC` is used as `MDCProvider` (https://github.com/quarkusio/quarkus/commit/65e153635589d6876a3eea175a80fdf4dcc3b09b#diff-99bb6be11b1e4b85ac938b26fbadf0ff1d58e23d88ab481ea5882df75ab3c18eR247). I use Uni to make parallel rest calls and eventually I clear the thread's MDC context but VertxMDC gets all MDC context and clears it. After uni calls, I lost my MDC info and logs are printed without MDC context.

```
Uni.createFrom().item(() -> DO_REST_CALL_HERE)
                .withContext((uni, ctx) -> {
                    ctx.keys().forEach(k -> MDC.put(k, ctx.get(k)));
                    return uni;
                }).eventually(MDC::clear).runSubscriptionOn(executor);
```
Do you have any suggestion to solve this problem?
